e2e/verifiable-credentials.service.e2e.ts

```diff
@@ -425,12 +426,48 @@ describe('Verifiable credentials tests', () => {
       });
       expect(selectResults.verifiableCredential).toHaveLength(1);
     });
-    test('filterSelfSignDescriptors() should return all input_descriptors that are not self-sign', () => {
-      expect(
-        verifiableCredentialsService.filterSelfSignDescriptors(descriptors)
-      ).toHaveLength(1);
+    test('initiateExchange() should filter self-sign data input_descriptors before fetching credentials and selecting matches', async () => {
```

Comment:
Maybe it would worth just noting the rationale here someone to reduce confusion regarding this test in the future
```suggestion
    /*
     * For rationale for this behaviour, see the comment on `verifiable-credentials-base.filterSelfSignDescriptors()`
     */
    test('initiateExchange() should filter self-sign data input_descriptors before fetching credentials and selecting matches', async () => {
```